src/terrain/index.js

```diff
@@ -0,0 +1,157 @@
+const { getRandomTerrainFunction } = require('./terrainFunctions');
+const terrainGenerator = (boardBounds = { x: 100, y: 100 }, terrainBounds = { x: 5, y: 5 }, numberOfTerrainElements = 5) => {
+    let { mockArray, mockWidth, mockHeight } = createMockArray(boardBounds, terrainBounds);
+    const keyPoint = {
+        x: Math.floor(Math.random() * mockWidth),
+        y: Math.floor(Math.random() * mockHeight)
+    };
+    mockArray = generateTerrain(mockArray, keyPoint, numberOfTerrainElements, mockWidth, mockHeight);
+    return mockArray;
+};
+
+createMockArray = (boardBounds, terrainBounds) => {
+    const boardX = boardBounds.x;
+    const boardY = boardBounds.y;
+    const terrainX = terrainBounds.x;
+    const terrainY = terrainBounds.y;
+
+    const mockArray = [];
+    const mockWidth =  boardX/terrainX;
+    const mockHeight = boardY/terrainY;
+
+    const defaultInfo = {
+        filled: false,
+        north: true,
+        south: true,
+        east: true,
+        west: true
+    };
+
+    for(let i = 0; i < mockWidth; i++) {
+        mockArray[i] = [];
+        for (let j = 0; j < mockHeight; j++) {
+            const info = { ...defaultInfo };
+            if (i === 0) {
+                info.west = false;
+            } else if (i === mockWidth - 1) {
+                info.east = false;
+            }
+            if (j === 0) {
+                info.north = false;
+            } else if (j === mockHeight - 1) {
+                info.south = false;
+            }
+            mockArray[i].push(info);
+        }
+    }
+    return {
+        mockArray,
+        mockWidth,
+        mockHeight
+    };
+};
+
+const deepClone = (toCopy) => {
+    const toReturn = [];
+    toCopy.forEach((row, rowNum) => {
+        toReturn[rowNum] = [];
+        row.forEach(elem => {
+            toReturn[rowNum].push({ ...elem });
+        });
+    });
+    return toReturn;
+};
+
+
+const cleanUpBoard = (board) => {
+    for (let i = 0; i < board.length; i++) {
+        for (let j = 0; j < board[i].length; j++) {
+            board[i][j] = {
+                filled: board[i][j].filled,
+                north: board[i][j].north,
+                south: board[i][j].south,
+                east: board[i][j].east,
+                west: board[i][j].west
+            };
+        }
+    }
+};
+
+const verifyKeyPoint = (board, keyPoint, boardWidth, boardHeight) => {
+    const pointsVerified = [];
+    for (let x = 0; x < boardWidth; x++) {
+        for (let y = 0; y < boardHeight; y++) {
+            let test = { x, y };
+            if ((pointsVerified.findIndex(point => point.x === test.x && point.y === test.y) > -1) ||
```

Comment:
this is kind of gnarly, maybe we can pull this out into its own function to make it more clear exactly what this check does. it'd also make this easier to read